qutip/lattice.py

```diff
@@ -0,0 +1,814 @@
+# This file is part of QuTiP: Quantum Toolbox in Python.
+#
+#    Copyright (c) 2011 and later, The QuTiP Project.
+#    All rights reserved.
+#
+#    Redistribution and use in source and binary forms, with or without
+#    modification, are permitted provided that the following conditions are
+#    met:
+#
+#    1. Redistributions of source code must retain the above copyright notice,
+#       this list of conditions and the following disclaimer.
+#
+#    2. Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#
+#    3. Neither the name of the QuTiP: Quantum Toolbox in Python nor the names
+#       of its contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+#    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+#    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#    HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+
+
+__all__ = ['Lattice1d', 'cell_structures']
+
+from numpy import (exp, expm1, log, log10)
+from scipy.sparse import (_sparsetools, isspmatrix, isspmatrix_csr,
+                          csr_matrix, coo_matrix, csc_matrix, dia_matrix)
+from qutip.fastsparse import fast_csr_matrix, fast_identity
+from qutip import *
+import numpy as np
+from scipy.sparse.linalg import eigs
+
+try:
+    from matplotlib.pyplot import *
+    import matplotlib.pyplot as plt
+except:
+    pass
+
+
+def Hamiltonian_2d(base_h, inter_hop_x, inter_hop_y, nx_units=1, PBCx=0,
+                   ny_units=1, PBCy=0, cell_num_site=1, length_for_site=1):
+    """
+    Returns the Hamiltonian as a csr_matrix from the specified parameters for a
+    2d space of sites in the x-major format.
+
+    Parameters
+    ==========
+    base_h : numpy matrix
```

Comment:
numpy matrix should be numpy.array?

Reply:
Thanks. There were a few errors there. Fixed them.